librz/include/rz_il/rzil_opcodes.h

```diff
@@ -0,0 +1,447 @@
+};
+
+/**
+ *  \struct rzil_op_ite_t
+ *  \brief op structure for `ite` (bool -> 'a pure -> 'a pure -> 'a pure)
+ *
+ *  ite c x y is x if c evaluates to b1 else y.
+ */
+struct rzil_op_ite_t {
+	RzIL_BOOL void *condition; ///< index of BOOL condition
+	RzIL_PURE_VAL void *x; ///< index of RzILVal operand 1
+	RzIL_PURE_VAL void *y; ///< index of RzILVal operand 2
+};
+
+/**
+ *  \struct rzil_op_var_t
+ *  \brief op structure for `var` ('a var -> 'a pure)
+ *
+ *  var v is the value of the variable v.
+ */
+struct rzil_op_var_t {
+	RzIL_VAR const char *v; ///< name of variable, const one
+};
+
+/**
+ *  \struct rzil_op_and__t
+ *  \brief op structure for `and_` (bool -> bool -> bool)
+ *
+ *  and_ x y is a conjunction of x and y.
+ */
+struct rzil_op_and__t {
+	RzIL_BOOL void *x; ///< index of the BOOL operand
+	RzIL_BOOL void *y; ///< index of the BOOL operand
+};
+
+/**
+ *  \struct rzil_op_or__t
+ *  \brief op structure for `or_` (bool -> bool -> bool)
+ *
+ *  or_ x y is a disjunction of x and y.
+ */
+struct rzil_op_or__t {
+	RzIL_BOOL void *x; ///< index of the BOOL operand
+	RzIL_BOOL void *y; ///< index of the BOOL operand
+};
+
+/**
+ *  \struct rzil_op_inv_t
+ *  \brief op structure for `inv` (bool -> bool)
+ *
+ *  inv x inverts x.
+ */
+struct rzil_op_inv_t {
+	RzIL_BOOL void *x; ///< index of the BOOL operand
+	RzIL_BOOL void *ret; ///< index of store the BOOL result
+};
+
+/**
+ *  \struct rzil_op_load_t
+ *  \brief op structure for `load` (('a, 'b) mem -> 'a bitv -> 'b bitv)
+ *
+ *  load m k is the value associated with the key k in the memory m.
+ */
+struct rzil_op_load_t {
+	RzIL_MEM int mem; ///< index of the memory in VM (different from the temp_val_list)
+	RzIL_BITV void *key; ///< index of the RzILBitVector key (address)
+};
+
+/**
+ *  \struct rzil_op_store_t
+ *  \brief op structure for `store` (('a, 'b) mem -> 'a bitv -> 'b bitv -> ('a, 'b) mem)
+ *
+ *  store m k x a memory m in which the key k is associated with the word x.
+ */
+struct rzil_op_store_t {
+	RzIL_MEM int mem; ///< index of memory in VM
+	RzIL_BITV void *key; ///< index of the RzILBitVector key (address)
+	RzIL_BITV void *value; ///< index of the RzILVal value (data) to store
+};
+
+// TODO : a better way to map enum to string
+// Remember to add new opcode in rz_il_op2str
+// if you add a new one.
+typedef enum {
+	// Init
+	RZIL_OP_VAR,
+	RZIL_OP_UNK,
+	RZIL_OP_ITE,
+
+	// RzILBool
+	RZIL_OP_B0,
+	RZIL_OP_B1,
+	RZIL_OP_INV,
+	RZIL_OP_AND_,
+	RZIL_OP_OR_,
+
+	// RzILBitVector
+	RZIL_OP_INT,
+	RZIL_OP_MSB,
+	RZIL_OP_LSB,
+	RZIL_OP_NEG,
+	RZIL_OP_NOT,
+	RZIL_OP_ADD,
+	RZIL_OP_SUB,
+	RZIL_OP_MUL,
+	RZIL_OP_DIV,
+	RZIL_OP_SDIV,
+	RZIL_OP_MOD,
+	RZIL_OP_SMOD,
+	RZIL_OP_LOGAND,
+	RZIL_OP_LOGOR,
+	RZIL_OP_LOGXOR,
+	RZIL_OP_SHIFTR,
+	RZIL_OP_SHIFTL,
+	RZIL_OP_SLE,
+	RZIL_OP_ULE,
+	RZIL_OP_CAST,
+	RZIL_OP_CONCAT,
+	RZIL_OP_APPEND,
+	// ...
+
+	// Memory
+	RZIL_OP_LOAD,
+	RZIL_OP_STORE,
```

Comment:
Wouldn't `RZIL_OP_LOAD` actually be a bit vector op and `RZIL_OP_STORE` be an effect?

Reply:
No, the operand and effect are separated in this case, see the original implementation: 
- http://binaryanalysisplatform.github.io/bap/api/odoc/bap-core-theory/Bap_core_theory/Theory/Effect/Sort/index.html

Reply:
Interesting, where are the operands in this link?